python/rikai/types/geometry.py

```diff
@@ -60,67 +63,114 @@ def to_numpy(self) -> np.ndarray:
 
 
 class Box2d(ToNumpy):
-    """2-D Bounding Box.
+    """2-D Bounding Box, defined by ``(xmin, ymin, xmax, ymax)``
 
     Attributes
     ----------
-    x : float
-        X-coordinate of the center point of the box
-    y : float
-        Y-coordinate of the center point of the box
-    width : float
-        The width of the box
-    height : float
-        The height of the box
+    xmin : float
+        X-coordinate of the top-left point of the box.
+    ymin : float
+        Y-coordinate of the top-left point of the box.
+    xmax : float
+        X-coordinate of the bottm-right point of the box.
+    ymax : float
+        Y-coordinate of the bottm-right point of the box.
     """
 
     __UDT__ = Box2dType()
 
-    def __init__(self, x: float, y: float, width: float, height: float):
+    def __init__(self, xmin: float, ymin: float, xmax: float, ymax: float):
+        self.xmin = float(xmin)
+        self.ymin = float(ymin)
+        self.xmax = float(xmax)
+        self.ymax = float(ymax)
+
+    @classmethod
+    def from_center(
+        cls, center_x: float, center_y: float, width: float, height: float
```

Comment:
nit: consider adding some basic validation. Since these constructors all have the same input types, sometimes users might use the wrong one and it'll succeed silently but give the wrong result. Adding some basic validation (e.g., width and height > 0) isn't foolproof but if the user uses the wrong one on a whole column these will catch the vast majority of human errors.

Reply:
Sure will do.

Will using `assert` break prod system tho?  Are there sth like `guava.precondition` that can be turn off in prod in python?

Reply:
python has a -o flag